### PR TITLE
[TMVA][SOFIE] New developments in SOFIE for parsing ONNX files 

### PR DESCRIPTION
**Pull request by lmoneta**
# This Pull request add several new developments to the SOFIE ONNX parser

## Changes and fixes 

- Add Softmax, Identity and Concat operators 
- Add fusing of MatMul and Add in a Gemm operator
- Add support for generic input Batch Size in RModel
- Add support for un-order input ONNX graph and capability in teh ONNX parser to re-order correctly

### CODE REVIEW

File: tmva/sofie/src/RModel.cxx [hunk 1]
```diff
@@ -164,15 +170,36 @@ namespace SOFIE{
...
+      // convert parametric shape to a dimensional shape
+      if (fReadyInputTensorInfos.size() != fInputTensorNames.size()) {
+         if ( fReadyInputTensorInfos.size() + fInputTensorInfos.size() != fInputTensorNames.size())
+            throw std::runtime_error("TMVA-SOFIE: RModel::Initializes: invalid inputs");
+         for (auto & input : fInputTensorInfos) {
+            std::vector<size_t> shape;
+            shape.reserve(input.second.shape.size());
+            for (auto & d : input.second.shape){
+               if (d.isParam)
+                  shape.push_back(batchSize);
```
Comment (sitongan) [inline tmva/sofie/src/RModel.cxx:185]: While for most of the use cases, if there is an input shape parameter it is going to be the batch size, in ONNX standard there can be other possibilities. for e.g. a CNN model with variable input image size. Should we hard-code this behaviour here, or is this just a temporary measure?

Comment (lmoneta) [reply tmva/sofie/src/RModel.cxx:185]: Good point !
Is there a way to know from ONNX if the parametrised input is the batch size or something else ?

**Review by sitongan — APPROVED** [review]
lgtm

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2004/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2004/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2004/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

**Pull request merged**